bundles/AdminBundle/Resources/public/js/pimcore/helpers.js

```diff
@@ -2737,7 +2737,25 @@ pimcore.helpers.isValidPassword = function (pass) {
 };
 
 pimcore.helpers.getDeeplink = function (type, id, subtype) {
-    return Routing.generate('pimcore_admin_login_deeplink', {}, true) + '?' + type + "_" + id + "_" + subtype;
+    let target = type + "_" + id + "_" + subtype;
+    let url = Routing.generate('pimcore_admin_login_deeplink', {}, true) + '?' + target;
+
+    let response = Ext.Ajax.request({
+        method: 'POST',
+        async: false,
```

Comment:
We should avoid `async: false` here, as it is deprecated.

Reply:
Just tried to implement it without `async: false` but I haven't found any solution. Any hint on what I can try?

Reply:
weird, `async: false`  doesn't look `deprecated` but discouraged, or am i missing something? 
https://docs.sencha.com/extjs/7.0.0/modern/Ext.Ajax.html#cfg-async
https://docs.sencha.com/extjs/7.6.0/modern/Ext.Ajax.html#cfg-async

Reply:
It's not about ExtJS, but synchronous requests in general, see: https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/Synchronous_and_Asynchronous_Requests#synchronous_request

Reply:
@matthiashamacher you just need to implement a `success` callback, just as we do in other places: 
https://github.com/pimcore/pimcore/blob/11.x/bundles/AdminBundle/public/js/pimcore/asset/asset.js#L389

Reply:
I don't think it is possible to make this async. The Problem is that the deeplink is directly set to the button. Therefore we cannot update the value after it is initially set: https://github.com/pimcore/pimcore/blob/11.x/bundles/AdminBundle/public/js/pimcore/element/abstract.js#L296
At least I haven't found a way. Also tried to save the deeplink to the globalmanager and getting it from there but it still uses the initial value (false) where the result of the request is not there yet.

Reply:
what about getting and storing the route name parameter in `globalmanager` once and change `getDeepLink` as following

```
pimcore.helpers.getDeeplink = function (type, id, subtype) {
    let target = type + "_" + id + "_" + subtype;
    let url;
    try {
        url = Routing.generate('pimcore.globalmanager.get('custom_admin_route_name')', {}, true) + '?' + target;
    } catch(e){
        url = Routing.generate('pimcore_admin_login_deeplink', {}, true) + '?' + target;
    }

    return url;
};
```
There is some redundancy, the `let url = Routing.generate('pimcore_admin_login_deeplink', {}, true) + '?' + target;` and the fallback in `getDeepLink()` will end up returning the same value. 

With the ajax approach, it affects user experience, it may requires multiple syncronous calls if one has lots of extjs tabs open, and it does at least twice for each element, first when it loads the button with the dropdown to allow copy of id/fullpath/deeplink and again when showing the content after clicking the `Info` button
https://github.com/pimcore/pimcore/blob/59357ac83ca165f41a53536bae14f7c3fc70f9f3/bundles/AdminBundle/Resources/public/js/pimcore/element/abstract.js#L279-L280
https://github.com/pimcore/pimcore/blob/59357ac83ca165f41a53536bae14f7c3fc70f9f3/bundles/AdminBundle/Resources/public/js/pimcore/asset/asset.js#L488-L489

Reply:
ah, even by adding "expose" as true? like in the documentation eg.
```
myCustomAdminRoute:
    options:
        expose: true
```
or in annotations
```
* @Route("/my-custom-login-page", options={"expose"=true}, name="myCustomAdminRoute") ?
```

Reply:
Yes if you expose the route then it would work but as I have said, that would make the custom admin entry point logic useless. When the route is public it makes no sense to even configure it. At least that is my opinion.

Reply:
Just pushed an experimental PR https://github.com/pimcore/pimcore/pull/13365 that could help in turning into asyncronous. Hope it helps

Reply:
Thanks for the idea. Changed a little bit so only the deeplink button appears after the ajax call succeded not the whole metainfo button.